pkg/client/client.go

```diff
@@ -1113,11 +1121,44 @@ func (c *Client) CreateOrUpdateSecret(ctx context.Context, s *v1.Secret) error {
 
 	required := s.DeepCopy()
 	mergeMetadata(&required.ObjectMeta, existing.ObjectMeta)
-
+	// Check if the Secret has an owner reference to a Service, that carries
+	// the annotation with key
+	// service.beta.openshift.io/serving-cert-secret-name and the Secrets
+	// name as the value.
+	// This means that service-ca-operator controls and populates the two
+	// data fields tls.crt and tls.key. We want to retain those on updates
+	// if they existing and are not empty.
+	if c.maybeHasServiceCAData(ctx, required) {
+		if v, ok := existing.Data["tls.crt"]; ok && len(v) > 0 {
```

Comment:
(nit) I would have overwrite the `Data` field with the content from the existing secret.

Reply:
I went with the more specific approach here, since `service-ca-operator` only manages those fields too. If we retain the whole data field, we would potentially keep user content in the data field.